static/js/exclusive-content.js

```diff
@@ -0,0 +1,135 @@
+const WORKER = 'https://webmonetization.org/api/exclusive-content'
+
+export function exclusiveContent() {
+  const { data, paymentPointer } = gatherExclusive()
+
+  if (document.monetization !== undefined) {
+    setMeta(paymentPointer)
+
+    document.monetization.addEventListener('monetizationpending', () => {
+      document.querySelectorAll('.exclusiveMessage').forEach((element) => {
+        element.innerHTML =
+          '<p>⸻ ⏰ Waiting for Web Monetization to start to unlock this content. ⸻</p>'
+      })
+    })
+
+    document.monetization.addEventListener('monetizationstart', () => {
+      document.querySelectorAll('.exclusiveMessage').forEach((element) => {
+        element.innerHTML =
+          '<p>⸻ ⏰ Waiting for Web Monetization receipt verification to unlock this content. ⸻</p>'
+      })
+    })
+
+    document.monetization.addEventListener('monetizationprogress', (event) => {
+      if (data.length) {
+        unlockExclusiveContent(data.pop(), event.detail.receipt)
```

Comment:
If you leave and return to a tab that has successfully unlocked content, the `Enjoy your exclusive content!` message gets replaced with the `monetizationpending` message above (the content remains unlocked).
Maybe `exclusiveMessage` should only be allowed to modified if the content hasn't been unlocked?

Reply:
Good catch, I wasn't aware of that. But thinking about the implementation and decryption happening client side, it makes sense. 

I was planning on adding the messages to the input fields when creating the exclusive content in a later iteration but I could do that now, too.

Reply:
I made the change: https://github.com/WICG/webmonetization/blob/0c1c17e165f9b1ef73dcfa60bb848ab6905925b3/static/js/exclusive-content.js#L10